services/user/Chainmail/plugin/src/lib.rs

```diff
@@ -1,15 +1,28 @@
 #[allow(warnings)]
 mod bindings;
+mod errors;
 
-use bindings::exports::chainmail::plugin::api::{Error, Guest as API};
+use bindings::exports::chainmail::plugin::api::{Error, Guest as Api};
+use bindings::exports::chainmail::plugin::queries::{Guest as Query, Message};
+use bindings::host::common::server as CommonServer;
 use bindings::transact::plugin::intf as Transact;
+use errors::ErrorType::QueryResponseParseError;
 use psibase::fracpack::Pack;
-use psibase::services::chainmail;
 use psibase::AccountNumber;
+use serde::Deserialize;
 
 struct ChainmailPlugin;
 
-impl API for ChainmailPlugin {
+#[derive(Debug, Deserialize)]
+struct MessageSerde {
+    msg_id: String,
+    receiver: String,
+    sender: String,
+    subject: String,
+    body: String,
+}
```

Comment:
You can add 
```
[package.metadata.component.bindings]
derives = ["serde::Deserialize"]
```
To the plugin's `Cargo.toml` in order for the `Message` struct generated bindings to derive from `Deserialize`.